source/solvers/gls_sharp_navier_stokes.cc

```diff
@@ -1951,6 +1943,253 @@ GLSSharpNavierStokesSolver<dim>::copy_local_rhs_to_global_rhs(
+            "p_y", particles[i_particle].position[1]);
+          particles_information_table.set_precision("p_y", 12);
+          if (dim == 3)
+            {
+              particles_information_table.add_value(
+                "p_z", particles[i_particle].position[2]);
+              particles_information_table.set_precision("p_z", 12);
+            }
+
+          particles_information_table.add_value(
+            "v_x", particles[i_particle].velocity[0]);
+          particles_information_table.set_precision("v_x", 12);
+          particles_information_table.add_value(
+            "v_y", particles[i_particle].velocity[1]);
+          particles_information_table.set_precision("v_y", 12);
+
+          if (dim == 3)
+            {
+              particles_information_table.add_value(
+                "v_z", particles[i_particle].velocity[2]);
+              particles_information_table.set_precision("v_z", 12);
+            }
+
+          particles_information_table.add_value(
+            "f_x", particles[i_particle].forces[0]);
+          particles_information_table.set_precision("f_x", 12);
+          particles_information_table.add_value(
+            "f_y", particles[i_particle].forces[1]);
+          particles_information_table.set_precision("f_y", 12);
+
+          if (dim == 3)
+            {
+              particles_information_table.add_value(
+                "f_z", particles[i_particle].forces[2]);
+              particles_information_table.set_precision("f_z", 12);
+            }
+
+          if (dim == 3)
+            {
+              particles_information_table.add_value(
+                "omega_x", particles[i_particle].omega[0]);
+              particles_information_table.set_precision("omega_x", 12);
+              particles_information_table.add_value(
+                "omega_y", particles[i_particle].omega[1]);
+              particles_information_table.set_precision("omega_y", 12);
+            }
+          particles_information_table.add_value("omega_z",
+                                                particles[i_particle].omega[2]);
+          particles_information_table.set_precision("omega_z", 12);
+          if (dim == 3)
+            {
+              particles_information_table.add_value(
+                "T_x", particles[i_particle].torques[0]);
+              particles_information_table.set_precision("T_x", 12);
+              particles_information_table.add_value(
+                "T_y", particles[i_particle].torques[1]);
+              particles_information_table.set_precision("T_y", 12);
+            }
+          particles_information_table.add_value(
+            "T_z", particles[i_particle].torques[2]);
+          particles_information_table.set_precision("T_z", 12);
+        }
+      // Write the table in the checkpoint file.
+      particles_information_table.write_text(output);
+    }
+}
+
+template <int dim>
+void
+GLSSharpNavierStokesSolver<dim>::read_checkpoint()
+{
```

Comment:
Add an application_test for this, Just a restart of a single particle falling with a single iteration...